pkg/sql/conn_io.go

```diff
@@ -370,6 +370,8 @@ type CopyIn struct {
 	// CopyDone is decremented once execution finishes, signaling that control of
 	// the connection is being handed back to the network routine.
 	CopyDone *sync.WaitGroup
+	// Once is used to decrement CopyDone exactly once.
+	Once *sync.Once
```

Comment:
one thought i had is if this would be more clear with an embedded struct:
```
	CopyDone struct {
		*sync.WaitGroup
		*sync.Once
	}
```

(so that it's obvious the two should be used together